src/algorithms/peter_norving.py

```diff
@@ -0,0 +1,153 @@
+class PeterNorvig:
+
+    def __init__(self):
+        self.digits = '123456789'
+        self.rows = 'ABCDEFGHI'
+        self.colums = self.digits
+        self.squares = self.cross(self.rows, self.colums)
+        self.unitlist = [self.cross(self.rows, colum) for colum in
+                         self.colums] + [self.cross(row, self.colums)
+                for row in self.rows] + [self.cross(row_square,
+                colum_square) for row_square in ('ABC', 'DEF', 'GHI')
+                for colum_square in ('123', '456', '789')]
+        self.units = dict((square, [unit for unit in self.unitlist
+                          if square in unit]) for square in
+                          self.squares)
+        self.peers = dict((square, set(sum(self.units[square], []))
+                          - set([square])) for square in self.squares)
+
+    def cross(self, key_list, value_list):
```

Comment:
Since the first parameter represents the row positions and the second the column position, wouldn't be better to name them of this way, right now key and values list does no have meaning on the context of the class